lib/awful/widget/common.lua

```diff
@@ -36,6 +37,51 @@ function common.create_buttons(buttons, object)
     end
 end
 
+local function default_template()
+    local ib  = wibox.widget.imagebox()
+    local tb  = wibox.widget.textbox()
+    local bgb = wibox.container.background()
+    local tbm = wibox.container.margin(tb, dpi(4), dpi(4))
+    local ibm = wibox.container.margin(ib, dpi(4))
+    local l   = wibox.layout.fixed.horizontal()
+
+    -- All of this is added in a fixed widget
+    l:fill_space(true)
+    l:add(ibm)
+    l:add(tbm)
+
+    -- And all of this gets a background
+    bgb:set_widget(l)
+
+    return {
+        ib      = ib,
+        tb      = tb,
+        bgb     = bgb,
+        tbm     = tbm,
+        ibm     = ibm,
+        primary = l,
+    }
+end
+
+local function custom_template(args)
+    local l = base.make_widget_from_value(args.widget_template)
+
+    -- For now, only there is no better way. If template could implement this
+    -- function even if it doesn't use the declarative construct.
+    assert(l.get_children_by_id)
```

Comment:
Perhaps `assert(l.get_children_by_id, "The given widget template did not result in a layout with a 'get_children_by_id' method")`, or something like that?